annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java

```diff
@@ -588,6 +590,30 @@ protected boolean validateResource(String url) throws ParamValidationException {
 		return true;
 	}
 
+	/**
+	 * This method verifies if provided right is in a list of valid licenses
+	 * @param right The right provided in the input
+	 * @return true if provided right is in a list of valid licenses
+	 * @throws ParamValidationException
+	 */
+	protected boolean validateRights(String right) throws ParamValidationException {
+
+		// remove version from the right
+		String RIGHT_DELIMITER = "/";		
+		String[] rightElems = right.split(RIGHT_DELIMITER);
+		String[] normalizedRightElems = Arrays.copyOf(rightElems, rightElems.length-1);
+		String normalizedRight = String.join(RIGHT_DELIMITER, normalizedRightElems)+RIGHT_DELIMITER;
+		String transcriptionLicenses = getConfiguration().getTranscriptionsLicenses();
+		List<String> licensesList = Arrays.asList(transcriptionLicenses.split("\\s*,\\s*"));
+		Set<String> rights = new HashSet<String>(licensesList);
```

Comment:
Use Streams and laizy loading. create getAcceptedLicenceses method. 

Set<String> set = Stream.of("a", "b")
                        .collect(Collectors.toCollection(HashSet::new));

Reply:
stream and collection methods were implemented